### PR TITLE
set NODE_ENV to test in vitest runner

### LINKED ISSUE #4289: running with vitest not setting the `NODE_ENV` to `test`
**Summary**

vitest set NODE_ENV to test, and when running mutation it does not

Same behavior as jest
#3580

**Stryker environment**

```
npm ls | grep stryker
├── @stryker-mutator/core@7.0.2
├── @stryker-mutator/vitest-runner@7.0.2
```

```
vitest@0.32.0
```

**Your Environment**

| software         | version(s)
| ---------------- | -------
| node             | 16.18.1
| npm              | 8.19.2
| Operating System | macOs

### PR DESCRIPTION
**Pull request by rluvaton**
fix #4289

I would like to add tests but I'm not really sure how, there are a lot of configuration needed, and coping from the jest test is not really fit

### CODE REVIEW

File: packages/vitest-runner/src/vitest-test-runner.ts [hunk 1]
```diff
@@ -89,6 +89,7 @@ export class VitestTestRunner implements TestRunner {
   }
 
   private async run(testIds: string[] = []): Promise<DryRunResult> {
+    this.setEnv();
```
Comment (nicojs) [inline packages/vitest-runner/src/vitest-test-runner.ts:92]: I think this could be set in `.init()`, instead of `run`.

**Review by nicojs — CHANGES_REQUESTED** [review]
Thanks for this PR! 🙏

I think it's OK to add a unit test for this. When you set the env in `init`, you can simply add a unit test to it in "test/unit/vitest-runner.spec.ts".

**Comment by rluvaton** [comment]
I will try to update later this, week, thanks

**Comment by rluvaton** [comment]
Hey @nicojs , there are a lot of missing things in the `createVitestMock` like `projects` and `state`, is there a reason to create these unit tests?

I tried to avoid unit tests and focus on component tests to reduce the number of changes I need to do in the tests when I do a refactor

**Comment by rluvaton** [comment]
added the test and moved the set env to init

File: packages/vitest-runner/test/unit/vitest-runner.spec.ts [hunk 2]
```diff
@@ -44,4 +44,12 @@ describe(VitestTestRunner.name, () => {
       await expect(sut.dispose()).not.rejected;
     });
   });
+
+  it('should set the NODE_ENV environment variable to test in init', async () => {
+    expect(process.env.NODE_ENV).to.not.equal('test');
```
Comment (nicojs) [inline packages/vitest-runner/test/unit/vitest-runner.spec.ts:49]: Maybe it's better to delete it?

```js
delete process.env.NODE_ENV;
```

**Comment by rluvaton** [comment]
> > is there a reason to create these unit tests
> 
> Generally, everything should be tested as low in the testing pyramid as possible. If at all possible. In this case it is possible, so yes, probably a unit test.

I tend to disagree, component tests are much easier to write (as you come from the user point of view) and cover much more while also protecting you from changing the tests when doing refactor across units

(IMO)


Take a look at:
- https://github.com/testjavascript/nodejs-integration-tests-best-practices
- https://kentcdodds.com/blog/the-testing-trophy-and-testing-classifications

File: packages/vitest-runner/test/unit/vitest-runner.spec.ts (hunk 2 above)
Comment (rluvaton) [reply packages/vitest-runner/test/unit/vitest-runner.spec.ts]: agree, done

**Review by nicojs — APPROVED** [review]

**Comment by rluvaton** [comment]
Hope to contribute to this project again soon!

It was great working with you!

**Pull request merged**